drivers/s3.go

```diff
@@ -285,32 +294,40 @@ func (os *s3Session) saveDataPut(ctx context.Context, name string, data []byte,
 			metadata[k] = aws.String(v)
 		}
 	}
-	contentType := aws.String(os.getContentType(name, data))
+	data, contentType, err := os.peekContentType(name, data)
+	if err != nil {
+		return "", err
+	}
 
-	params := &s3.PutObjectInput{
-		Bucket:        bucket,  // Required
-		Key:           keyname, // Required
-		Metadata:      metadata,
-		Body:          bytes.NewReader(data),
-		ContentType:   contentType,
-		ContentLength: aws.Int64(int64(len(data))),
+	uploader := s3manager.NewUploader(os.s3sess, func(u *s3manager.Uploader) {
+		u.Concurrency = 8
```

Comment:
Consider extracting this magic number to some constants with the name. Or at least give here some comments about how you came to these numbers.